src/main/java/com/common/validation/utils/MaskingHelper.java

```diff
@@ -1,9 +1,20 @@
 package com.common.validation.utils;
 
+import java.util.List;
+import java.util.regex.Pattern;
+
 public class MaskingHelper {
     //MASK Profane words
     public static String maskProfanity(String input, String pattern) {
-        input = input.replaceAll(pattern, "******");
+        input = input.replaceAll("\\b" + pattern + "\\b", "******");
+        return input;
+    }
+
+    public static String maskAllProfanity(String input, List<String> wordList) {
+        for (String word : wordList) {
+            String regex = "\\b" + Pattern.quote(word) + "\\b";
+            input = input.replaceAll(regex, "***");
```

Comment:
Seems like the code is the same, why don't just simply invoking the maskProfanity() above?